pkg/resources/management.cattle.io/v3/roletemplate/validator.go

```diff
@@ -192,6 +192,9 @@ func validateCreateFields(newRole *v3.RoleTemplate, fldPath *field.Path) *field.
 }
 
 func validateContextValue(newRole *v3.RoleTemplate, fldPath *field.Path) *field.Error {
+	if newRole.Context == clusterContext && newRole.ProjectCreatorDefault == true {
```

Comment:
In the docs you stated

> If the `roletemplate.ProjectCreatorDefault` is true, context must equal `"project"`

But this checks only that the context is `"cluster"`. Context can be `"cluster"` `"project"` or `""` ([source](https://github.com/rancher/rancher/blob/c5b4413739fb7a9f935eadeb7723b28b90ee22be/pkg/apis/management.cattle.io/v3/authz_types.go#L280)), so this check should probably be:
```suggestion
	if newRole.Context != projectContext && newRole.ProjectCreatorDefault == true {
```

Alternatively, the doc should say 

> If the `roletemplate.ProjectCreatorDefault` is true, context can not be `"cluster"`